### ISSUE TITLE
a problem in using nvda 2015.4 and later versions with some media players.

### ISSUE THREAD

**Issue by zahra21** [nvaccess/nvda#6513]
hello every one.
my favorite media player is potplayer.
everythings was okay before release of nvda 2015.4 for me.
when i installed nvda 2015.4 and also all later versions, i observed a big problem.

steps to reproduce.
1- select many files or one audio folder to use with potplayer.
2- play with potplayer.
3- press alt+f4 to close the player.
current result: after exiting the player, nvda says unknown, does not read menus, documents, etc and you should press alt+control+n to restart nvda for use.
or if this does not  happen after the first file immediately, this occur eventually after playing many files and exiting the program!
expected result: nvda should works as usual and does not stop reading after exiting media players!
i only use potplayer, but someone told me that this happen for him in using windows media player!
i wish that you fix it in nvda 2016.4
i appreciate your great screen reader, pray for you and request divine mercy for you.

**Comment by Brian1Gaff** [comment]
If this is in xp, it could well be something to do with lack of UIA perhaps?

As I say, its hard to comment on something you cannot duplicate yourself.

Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
----- Original Message -----

**Comment by zahra21** [comment]
> hi brian.
> it occurs in different windows versions, not only xp.
> the person who told me that he face this problem with windows media player, uses windows 7!

**Comment by feerrenrut** [comment]
@zahra21 thanks for filing the issue, as it stands we need a little more information to progress this. 
In order for us to investigate and identify the issue we will need access to your NVDA log.

First you will need to set your log level to `Debug` (in the General Settings dialog), and then reproduce the problem using potplayer. For information on how to collect the logs see the [Log files and crash dumps](https://github.com/nvaccess/nvda/wiki/LogFilesAndCrashDumps). Then attach the log to this issue.

**Comment by zahra21** [comment]
> hi feerrenrut.
> thanks extremely for your help.
> i studied your link, made log file for you and now send it for you.
> but i am 100 percent sure that this problem is in nvda 2015.4 and later versions for me!
> [my nvda log.txt](https://github.com/nvaccess/nvda/files/555052/my.nvda.log.txt)

**Comment by feerrenrut** [comment]
@zahra21 thanks for taking the time to follow this up and attaching the log file. I have had a look at it, but it seems to be missing the information we require. Because nvda is restarted while reproducing this issue, we really need the `nvda-old.log`. Could you please try the following for me:

- Set the log level to "Debug"
- Reproduce the issue (as you described, pressing `alt-f4` while potplayer is playing)
- Navigate to `%temp%` and copy both `nvda-old.log` and `nvda.log`. For more information on how to do this see [Log files and crash dumps](https://github.com/nvaccess/nvda/wiki/LogFilesAndCrashDumps)
- Attach both of these files to this issue as you did previously.

**Comment by zahra21** [comment]
hi again.
testing it very easy.
you can test it with potplayer and windows media player according to my steps to reproduce the issue!
nvda did not restart.
i restarted it myself, because it did not read the menus and anythings.
i did not have old log to attach it here.

**Comment by zahra21** [comment]
hello every one.
i have a good news!
i tested version 1.6.63840 and the issue persists.
but after testing potplayer 1.6.63891 which is currently the last stable version, it seems that the problem is completely solved for me!
i realy appreciate the programmers who tried to solve this big issue!
thanks extremely, i pray for you in my daily prayers and request divine extreme mercy for you!

**Comment by fernando-jose-silva** [comment]
Good afternoon.
Dear, I still have this problem when using windows media player on windows 10 and on windows 7.
To reproduce:
Open an audio file in windows media player.
Walk with tab to the "search" slider.
Press right arrow to advance the audio file.
Now press alt + f4 to close windows media player.
Expected result is that nvda still works without an error.
Result that hits, when trying to navigate through any window, windows explorer, start menu, desktop, nvda does not report on the focused items, and emits an error sound.
Follow the log:

Input: kb(laptop):alt+f4
ERROR - eventHandler.executeEvent (17:37:08):
error executing event: gainFocus on <NVDAObjects.IAccessible.IAccessible object at 0x06639150> with extra args of {}
Traceback (most recent call last):
  File "eventHandler.pyc", line 143, in executeEvent
  File "eventHandler.pyc", line 91, in __init__
  File "eventHandler.pyc", line 98, in next
  File "C:\Users\a50596\AppData\Roaming\nvda\addons\enhancedTouchGestures\globalPlugins\enhancedTouchGestures.py", line 84, in event_gainFocus
  File "eventHandler.pyc", line 98, in next
  File "C:\Users\a50596\AppData\Roaming\nvda\addons\remote\globalPlugins\remoteClient\__init__.py", line 341, in event_gainFocus
  File "eventHandler.pyc", line 98, in next
  File "C:\Users\a50596\AppData\Roaming\nvda\addons\focusHighlight\globalPlugins\focusHighlight.py", line 416, in event_gainFocus
  File "eventHandler.pyc", line 98, in next
  File "C:\Users\a50596\AppData\Roaming\nvda\addons\easyTableNavigator\globalPlugins\easyTableNavigator.py", line 82, in event_gainFocus
  File "eventHandler.pyc", line 98, in next
  File "C:\Users\a50596\AppData\Roaming\nvda\addons\AudioThemes\globalPlugins\AudioThemes\__init__.py", line 63, in event_gainFocus
  File "C:\Users\a50596\AppData\Roaming\nvda\addons\AudioThemes\globalPlugins\AudioThemes\utiles.py", line 44, in getLocation
AttributeError: 'NoneType' object has no attribute 'role'
ERROR - eventHandler.executeEvent (17:37:08):
error executing event: gainFocus on <NVDAObjects.Dynamic_UIItemListItemUIA object at 0x06639350> with extra args of {}
Traceback (most recent call last):
  File "eventHandler.pyc", line 138, in executeEvent
  File "eventHandler.pyc", line 151, in doPreGainFocus
  File "api.pyc", line 134, in setFocusObject
  File "appModuleHandler.pyc", line 268, in handleAppSwitch
AttributeError: 'AppModule' object has no attribute '_configProfileTrigger'
IO - inputCore.InputManager.executeGesture (17:37:13):
Input: kb(laptop):downArrow
ERROR - eventHandler.executeEvent (17:37:13):
error executing event: gainFocus on <NVDAObjects.Dynamic_UIItemListItemUIA object at 0x0B55EC30> with extra args of {}
Traceback (most recent call last):
  File "eventHandler.pyc", line 138, in executeEvent
  File "eventHandler.pyc", line 151, in doPreGainFocus
  File "api.pyc", line 134, in setFocusObject
  File "appModuleHandler.pyc", line 268, in handleAppSwitch
AttributeError: 'AppModule' object has no attribute '_configProfileTrigger'
IO - speech.speak (17:37:14):
Speaking [u'Maid with the Flaxen Hair.mp3\nTipo de item: Som no Formato MP3\nTamanho: 3,92 MB  dica de ferramenta']
IO - inputCore.InputManager.executeGesture (17:37:22):
Input: kb(laptop):NVDA+f1
INFO - globalCommands.GlobalCommands.script_navigatorObject_devInfo (17:37:22):
Developer info for navigator object:
name: None
role: ROLE_UNKNOWN
states: STATE_INVISIBLE
isFocusable: False
hasFocus: False
Python object: <NVDAObjects.IAccessible.IAccessible object at 0x06639150>
Python class mro: (<class 'NVDAObjects.IAccessible.IAccessible'>, <class 'NVDAObjects.window.Window'>, <class 'NVDAObjects.NVDAObject'>, <class 'baseObject.ScriptableObject'>, <class 'baseObject.AutoPropertyObject'>, <type 'object'>)
description: None
location: None
value: None
appModule: <'appModuleHandler' (appName u'[system process]', process ID 0) at address 6639310>
appModule.productName: exception: processHandle is 0
appModule.productVersion: exception: processHandle is 0
TextInfo: <class 'NVDAObjects.NVDAObjectTextInfo'>
windowHandle: 919694
windowClassName: u''
windowControlID: 0
windowStyle: 0
windowThreadID: 0
windowText: u''
displayText: exception: 'NoneType' object is not iterable
IAccessibleObject: <POINTER(IAccessible) ptr=0x87958c8 at 6626170>
IAccessibleChildID: 0
IAccessible event parameters: windowHandle=919694, objectID=-4, childID=0
IAccessible accName: exception: (-2147023174, 'O servidor RPC n\xe3o est\xe1 dispon\xedvel.', (None, None, None, 0, None))
IAccessible accRole: exception: (-2147023174, 'O servidor RPC n\xe3o est\xe1 dispon\xedvel.', (None, None, None, 0, None))
IAccessible accState: exception: (-2147023174, 'O servidor RPC n\xe3o est\xe1 dispon\xedvel.', (None, None, None, 0, None))
IAccessible accDescription: exception: (-2147023174, 'O servidor RPC n\xe3o est\xe1 dispon\xedvel.', (None, None, None, 0, None))
IAccessible accValue: exception: (-2147023174, 'O servidor RPC n\xe3o est\xe1 dispon\xedvel.', (None, None, None, 0, None))

**Comment by josephsl** [comment]
Hi, what happens if you disable add-ons? Thanks.

**Comment by jcsteh** [comment]
@zahra21, thanks for following this up.

The Windows Media player problem is covered by #5467. I'll close this issue now; please see #5467 for the Media Player issue. Thanks.

**Comment by zahra21** [comment]
hi all.
thanks so much for helping me.
and i asked someone confirms that he had this problem using windows
media player on windows 7.
i dont know the version of windows media player, but he tested this
with different versions of nvda and the result was only crash of nvda.
for me, in previous versions of potplayer, nvda said unknown and
became completely silent.
i used control+alt+n to operate it again.
could you please test it and see if the problem is resulved for you
using potplayer 1.6.63891?
thanks extremely and God bless you all.

On 1/19/17, James Teh <notifications@github.com> wrote:
> @zahra21, thanks for following this up.
>
> The Windows Media player problem is covered by #5467. I'll close this issue
> now; please see #5467 for the Media Player issue. Thanks.
>
> --
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/nvaccess/nvda/issues/6513#issuecomment-273642105


-- 
we have not sent you but as a mercy to the creation.
holy quran, chapter 21, verse 107.
in the very authentic narration is:
imam hosein is the beacon of light and the ark of salvation.
best website for studying islamic book in different languages
al-islam.org

**Comment by zahra21** [comment]
hi.
yesterday i downloaded nvda 2017.1 and today downloaded the potplayer 1.6.63840 to test the result.
fortunately everythings are very great for me.
could you please add Milestone nvda 2017.1 for this bug to indicate its fixed completely?
thanks extremely every one for your help, i sincerely pray for you in my daily prayers and request divine extreme and infinite mercy and graces for you.
God bless you all for this great software nvda!
